### PR TITLE
Fix the build on Linux when system libraries are used

### PR DESCRIPTION
**Pull request by crsib**
Fixes a problematic `get_package_interface` which was modifying a global list `LIBRARIES`.

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

File: cmake-proxies/cmake-modules/AudacityFunctions.cmake [hunk 1]
```diff
@@ -51,22 +51,28 @@ function( set_dir_folder dir folder)
...
+      find_library( LIB_${lib} ${lib} HINTS ${package_linkdirs} )
+      list( APPEND package_libraries ${LIB_${lib}} )
    endforeach()
-endmacro()
+
+   # And add it to our target
+   target_include_directories( ${target} INTERFACE ${package_includes} )
+   target_link_libraries( ${target} INTERFACE ${package_libraries} )
+
+   message(STATUS "Inteface ${target}:\n\tinclude: ${includes}\n\tLibraries: ${LIBRARIES}")
```
Comment (Paul-Licameli) [inline cmake-proxies/cmake-modules/AudacityFunctions.cmake:74]: “Interface”

**Review by Paul-Licameli — APPROVED** [review]
Two suggestions

File: cmake-proxies/cmake-modules/AudacityFunctions.cmake [hunk 2]
```diff
@@ -51,22 +51,28 @@ function( set_dir_folder dir folder)
...
       ${${package}_INCLUDE_DIRS}
    )
 
-   set( LINKDIRS
+   set( package_linkdirs
       ${${package}_LIBDIR}
    )
 
    # We resolve the full path of each library to ensure the
    # correct one is referenced while linking
```
[the remark this answers is not in the document]
Comment (crsib) [reply cmake-proxies/cmake-modules/AudacityFunctions.cmake:64]: Will do!

File: cmake-proxies/cmake-modules/AudacityFunctions.cmake (hunk 1 above)
Comment (crsib) [reply cmake-proxies/cmake-modules/AudacityFunctions.cmake:74]: Fixed

**Pull request merged**